### PR TITLE
feat(react): rewire webpack config

### PR DESCRIPTION
**Pull request by iliapolo**
Support basic react property overrides without needing to eject.

Based on https://www.npmjs.com/package/react-app-rewired.

You can see an example of this mechanism here: https://github.com/cdklabs/construct-hub-webapp/blob/main/config-overrides.js

Once this is merged, we can migrate construct-hub-webapp to use it. 

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/web/react.ts [hunk 1]
```diff
@@ -150,29 +187,50 @@ export class ReactComponent extends Component {
...
+      const configOverrides = new SourceCode(this.project, 'config-overrides.js');
+      configOverrides.line(`// ${FileBase.PROJEN_MARKER}`);
+      configOverrides.line('/**');
+      configOverrides.line(' * Override CRA configuration without needing to eject.');
+      configOverrides.line(' *');
+      configOverrides.line(' * @see https://www.npmjs.com/package/react-app-rewired');
+      configOverrides.line(' */');
+      configOverrides.open('module.exports = function override(config, env) {');
+      for (const [key, value] of Object.entries(rewire)) {
+        configOverrides.line(`config.${key} = ${JSON.stringify(value)}`);
```
Comment (eladb) [inline src/web/react.ts:202]:
```suggestion
        configOverrides.line(`config.${key} = ${JSON.stringify(value)};`);
```

File: src/web/react.ts [hunk 2]
```diff
@@ -150,29 +187,50 @@ export class ReactComponent extends Component {
...
+      }
+      configOverrides.line('return config');
```
Comment (eladb) [inline src/web/react.ts:204]:
```suggestion
      configOverrides.line('return config;');
```

File: src/web/react.ts [hunk 3]
```diff
@@ -150,29 +187,50 @@ export class ReactComponent extends Component {
...
+      configOverrides.close('}');
```
Comment (eladb) [inline src/web/react.ts:205]:
```suggestion
      configOverrides.close('};');
```

File: src/web/react.ts [hunk 4]
```diff
@@ -150,29 +187,50 @@ export class ReactComponent extends Component {
       project.addDevDeps('@types/jest', '@types/node', '@types/react', '@types/react-dom');
     }
 
+    const rewire = options.rewire;
+
+    if (rewire) {
+      const configOverrides = new SourceCode(this.project, 'config-overrides.js');
```
Comment (eladb) [inline src/web/react.ts:193]: Can this go under `.projen/react-config-overrides.js` to avoid polluting the root?

File: src/web/react.ts [hunk 5]
```diff
@@ -150,29 +187,50 @@ export class ReactComponent extends Component {
...
+      configOverrides.line(' */');
+      configOverrides.open('module.exports = function override(config, env) {');
+      for (const [key, value] of Object.entries(rewire)) {
+        configOverrides.line(`config.${key} = ${JSON.stringify(value)}`);
+      }
+      configOverrides.line('return config');
+      configOverrides.close('}');
+    }
+
+    const reactScripts = rewire ? 'react-app-rewired' : 'react-scripts';
```
Comment (eladb) [inline src/web/react.ts:208]: Don't we need to install this dependency?

**Review by eladb — CHANGES_REQUESTED** [review]

File: src/web/react.ts (hunk 4 above)
Comment (iliapolo) [reply src/web/react.ts:193]: Looks like its [possible](https://github.com/timarney/react-app-rewired/commit/73596cc59e05ae7b0ec42b4332fca448bf30134b). I'll try it out.

File: src/web/react.ts (hunk 5 above)
Comment (iliapolo) [reply src/web/react.ts:208]: Whoops! Good catch...

File: src/web/react.ts (hunk 4 above)
Comment (iliapolo) [reply src/web/react.ts]: Moved 👍

**Review by eladb — APPROVED** [review]

**Pull request merged**